src/App.jsx

```diff
@@ -1,12 +1,74 @@
 import './App.css'
+import {useEffect, useState} from "react";
+import axios from "axios";
+import PokemonCard from "./components/pokemonCard/PokemonCard.jsx";
 
 function App() {
+    const [pokemonList, setPokemonList] = useState([])
+    const [loading, setLoading] = useState(false);
+    const [currentPage, setCurrentPage] = useState(0);
+    const [error, setError] = useState(null);
 
-  return (
-    <>
-      <h1>Gotta catch em all!</h1>
-    </>
-  )
+    useEffect(() => {
+        const controller = new AbortController();
+
+        async function fetchData() {
+            setError(null);
+            setLoading(true);
+            setPokemonList([]);
+            try {
+                const response = await axios.get(`https://pokeapi.co/api/v2/pokemon?offset=${currentPage * 20}&limit=20`, {
+                    signal: controller.signal
+                });
+                const pokemonDetails = await Promise.all(response.data.results.map(async (pokemon) => {
+                    const detailsResponse = await axios.get(pokemon.url);
+                    return {
+                        name: pokemon.name,
+                        image: `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${detailsResponse.data.id}.png`,
+                        moves: detailsResponse.data.moves.length,
+                        weight: detailsResponse.data.weight,
+                        abilities: detailsResponse.data.abilities.map(ability => ability.ability.name)
+                    };
+                }));
+                setPokemonList(pokemonDetails);
+                console.log("Pokemon data:", response.data.results);
+            } catch (error) {
+                console.error("Error fetching Pokémon data:", error);
+                setError("Failed to load Pokémon data. Please try again later.");
+            } finally {
+                setLoading(false);
+            }
+        }
+
+        fetchData();
```

Comment:
Omdat `currentPage`, waar deze `useEffect` op gebasseerd is, ook eerst veranderd van "niks" naar de initiele waarde, is het goed om hier `currentPage && fetchData()` van te maken zodat dit echt pas na de mount voor de eerste keer kan gebeuren!